src/desktop/apps/fair/fairRedirection.ts

```diff
@@ -0,0 +1,60 @@
+// @ts-ignore
+import { DISABLE_FAIRS_UPDATE_SLUGS, ENABLE_FAIRS_UPDATE } from "../../config"
+
+export const redirectFairRequests = (req, res, next) => {
+  const isFairProfile = res.locals?.profile?.get("owner_type") === "Fair"
+  if (!isFairProfile) return next()
+
+  // TODO: some time after launch of the updated fairs experience
+  // this conditional can be reduced to just the res.redirect
+  if (shouldRedirectToNewFairPage(req)) {
+    res.redirect(301, redirectPath(req, res))
+  } else {
+    next()
+  }
+}
+
+const shouldRedirectToNewFairPage = req => {
+  if (isNewFairPageDisabledForThisFair(req)) return false
+  if (shouldUserSeeNewFairPage(req)) return true
+
+  return false
+}
+
+const isNewFairPageDisabledForThisFair = req => {
+  return !!DISABLE_FAIRS_UPDATE_SLUGS?.split(/\s*,\s*/).includes(req.params.id)
+}
+
+const shouldUserSeeNewFairPage = req => {
+  const userHasLabFeature = !!req.user?.hasLabFeature("Fairs experience update")
+  const featureIsEnabledSitewide = ENABLE_FAIRS_UPDATE
+
+  return userHasLabFeature || featureIsEnabledSitewide
+}
+
+const redirectPath = (req, res) => {
+  const fairId = res.locals.profile?.get("owner")?.id
+  const defaultPath = `/fair/${fairId}`
+
+  try {
```

Comment:
It seems nice to have this in a `try`, but I'm not sure what would be expected to error, is this more for just: in case there's some scenario where `req.locals` is `undefined`, or something like that? I think those would always be guaranteed to exist (but not sure!). At least if there wasn't a `try` here, any errors would show up in Sentry and should be pretty diagnosable from there (since that would include the URL).

Reply:
This was a hedge against my fuzzy understanding of all the routing logic pertaining to fairs 😅. But I think you're right, an honest-to-goodness error might be preferable here.